### DISCUSSION TITLE
MLEngineStartTrainingJobOperator is not working on the pytorch container images of Google officials provide method.

### DISCUSSION THREAD

**Discussion by roymondliao** [apache/airflow#17119] [Q&A]
According to #13318 and #13192 to provide using the custom images at MLEngineStartTrainingJobOperator but that has conflict at using pytorch container images on ML Engine (https://cloud.google.com/ai-platform/training/docs/getting-started-pytorch#gpu_1).

The document example on below. When using the the imageURI that provide by Google is required command with `module-name`(`training_python_module` in operator) and `package-path`(`package_uris` in operator).

But the code at line [1221](https://github.com/apache/airflow/blob/9b3bfe67019f4aebd37c49b10c97b20fa0581be1/airflow/providers/google/cloud/operators/mlengine.py#L1221) is conflict with the rule.

```
gcloud ai-platform jobs submit training ${JOB_NAME} \
  --region=us-central1 \
  --master-image-uri=gcr.io/cloud-ml-public/training/pytorch-gpu.1-7 \
  --scale-tier=CUSTOM \
  --master-machine-type=n1-standard-8 \
  --master-accelerator=type=nvidia-tesla-p100,count=1 \
  --job-dir=${JOB_DIR} \
  --package-path=./trainer \
  --module-name=trainer.task \
  -- \
  --train-files=gs://cloud-samples-data/ai-platform/chicago_taxi/training/small/taxi_trips_train.csv \
  --eval-files=gs://cloud-samples-data/ai-platform/chicago_taxi/training/small/taxi_trips_eval.csv \
  --num-epochs=10 \
  --batch-size=100 \
  --learning-rate=0.001
```

Does any suggestion ?

**Comment by leahecole** [comment]
Can you share more about your DAG configuration, the error you're seeing and the output you expected to see? And if those outputs/errors are in Airflow vs. in AI Platform? That might help us debug a bit more. It is entirely possible there could be a bug in the operator, in which case PRs are always welcome 😄 but I'm not 100% sure and am not the most familiar with this operator in particular.

**Reply by roymondliao** [reply]
Follow by the GCP AI Platform document [create a custom container ](https://cloud.google.com/ai-platform/training/docs/custom-containers-training#create_a_custom_container) that the container must set **ENTRYPOINT**  in the docker file, but the [getting start with Pytorch](https://cloud.google.com/ai-platform/training/docs/getting-started-pytorch#gpu_1) is still needed to set the `package_uris` and `training_python_module` to provide the ml engine run the module.

The problem is the GCP AI Platform provide Pytorch container allow you train job, but still need to follow the original launch rule and the way of custom container is not following this rule.

So, at code [line 1221](https://github.com/apache/airflow/blob/76e6315473671b87f3d5fe64e4c35a79658789d3/airflow/providers/google/cloud/operators/mlengine.py#L1221) condition is only satisfy the custom container way, but not fit the way for launching the Pytorch container.
```
if (package_uris or training_python_module) and custom_image:
         raise AirflowException(
                'Either a Python package with a Python module or '
                'a custom Docker image should be provided but not both.'
            )
```
Btw, I solved this problem, just comment out the code above.

My testing env and operator setting on below:

Image: airflow:2.0.2 (modified by https://airflow.apache.org/docs/apache-airflow/stable/docker-compose.yaml)
Operator setting:
```
MLEngineStartTrainingJobOperator(
	task_id=//task_id
        job_id=//job_id,
        project_id=//project_id
        package_uris=[//package_uris],
        training_python_module="trainer.task",
        training_args=[
            '--job-dir', //job-dir,
        ],
        region="asia-east1",
        master_config={
        	'imageUri': "gcr.io/cloud-ml-public/training/pytorch-gpu.1-7"
        	},
        scale_tier="CUSTOM"
        master_type="standard_p100"
        python_version="3.7"
        gcp_conn_id=//gcp_conn_id
    )
```

**Reply by potiuk** [reply]
> So, at code line 1221 condition is only satisfy the custom container way, but not fit the way for launching the Pytorch container.

Can you please submit PR fixing it maybe ?

**Reply by leahecole** [reply]
Thanks so much for sharing further info! +1 to Jarek's suggestion
